### ISSUE TITLE
r2 uses 100% CPU while waiting for user input

### ISSUE THREAD

**Issue by bennyty** [radare/radare2#6944]
radare2 1.3.0-git 1 @ linux-x86-64 git.1.3.0-git
commit: 5d669ed2f8be5734d7ebd6b34d203d777c40c5e7 build: 2017-03-07__16:04:04

The server that is running r2 is also running a daemon that will kill any process using ~100% cpu for more than 30 seconds.

After `dc` if I do not finish passing input to the process within 30 seconds r2 will be killed; gdb does not eat up 100% cpu while waiting for input.

Why does r2 use 100% cpu?

**Comment by radare** [comment]
Never seen that behaviour. R2 doesnt uses threads, so 100% cpu usage means non interactive.. what do you do to reproduce this? You can use gdb to debug r2 and tell us why this happens

**Comment by bennyty** [comment]
I am trying to find a cause. I was able to reproduce this independently on a different machine with a different binary. Open a simple binary with `r2 -d`. Use `dc` to run the program and don't type any input into the prompt. Check `top` and notice that r2 is suddenly using >90% cpu.

This is the backtrace I have
```
0x00007f3d0d5704ca in __GI___waitpid (pid=0x3a60, stat_loc=0x7ffc4c9417d4,
    options=0x40000001) at ../sysdeps/unix/sysv/linux/waitpid.c:29
29      ../sysdeps/unix/sysv/linux/waitpid.c: No such file or directory.
gdb-peda$ bt
#0  0x00007f3d0d5704ca in __GI___waitpid (pid=0x3a60, stat_loc=0x7ffc4c9417d4,
    options=0x40000001) at ../sysdeps/unix/sysv/linux/waitpid.c:29
#1  0x00007f3d105d1ab1 in linux_dbg_wait (dbg=0x56239bebf570, pid=0x3a60)
    at p/native/linux/linux_debug.c:317
#2  0x00007f3d105cedee in r_debug_native_wait (dbg=0x56239bebf570, pid=0x3a60)
    at p/debug_native.c:360
#3  0x00007f3d105c4eea in r_debug_wait (dbg=0x56239bebf570, bp=0x7ffc4c941890)
    at debug.c:588
#4  0x00007f3d105c5ded in r_debug_continue_kill (dbg=0x56239bebf570, sig=0x0)
    at debug.c:934
#5  0x00007f3d105c618c in r_debug_continue (dbg=0x56239bebf570) at debug.c:1026
#6  0x00007f3d113ee3f5 in cmd_debug_continue (core=0x56239b7b14e0 <r>,
    input=0x56239befe831 "c") at cmd_debug.c:3111
#7  0x00007f3d113effb0 in cmd_debug (data=0x56239b7b14e0 <r>,
    input=0x56239befe831 "c") at cmd_debug.c:3598
#8  0x00007f3d114876d5 in r_cmd_call (cmd=0x56239bebc240,
    input=0x56239befe830 "dc") at cmd_api.c:226
#9  0x00007f3d1144bc09 in r_core_cmd_subst_i (core=0x56239b7b14e0 <r>,
    cmd=0x56239befe830 "dc", colon=0x0) at cmd.c:2066
#10 0x00007f3d114491e3 in r_core_cmd_subst (core=0x56239b7b14e0 <r>,
    cmd=0x56239befe830 "dc") at cmd.c:1341
#11 0x00007f3d1144db2e in r_core_cmd (core=0x56239b7b14e0 <r>,
    cstr=0x56239bfdf2f0 "dc", log=0x1) at cmd.c:2622
#12 0x00007f3d113d7d4e in r_core_prompt_exec (r=0x56239b7b14e0 <r>)
    at core.c:1878
#13 0x000056239b5aeff6 in main (argc=0x3, argv=0x7ffc4c943f68,
    envp=0x7ffc4c943f88) at radare2.c:1164
#14 0x00007f3d0d4c5830 in __libc_start_main (main=0x56239b5ac944 <main>,
    argc=0x3, argv=0x7ffc4c943f68, init=<optimized out>, fini=<optimized out>,
    rtld_fini=<optimized out>, stack_end=0x7ffc4c943f58)
    at ../csu/libc-start.c:291
#15 0x000056239b5abd29 in _start ()
```

**Comment by radare** [comment]
ok i can reproduce that. let me fix it

**Comment by radare** [comment]
cc @leberus

**Comment by leberus** [comment]
i take a look

**Comment by radare** [comment]
Fixed in 91016693cb7ca4b545c3067871e514059e226c47

but feel free to test or enhance
